test/utils/helpers.go

```diff
@@ -376,6 +401,42 @@ func GenerateNameFromText(fullText string, focus []string) string {
 	return strings.ReplaceAll(fullText, " ", "")
 }
 
+// SetAgentConfig configures the cluster, generates agent kubeconfigs and sets terraform env vars.
+func SetAgentConfig(c client.Client, ns *corev1.Namespace, cloudProviders, antreaVersion, kubeconfig, dir string) error {
+	err := c.Create(context.TODO(), ns)
+	if err != nil {
+		return fmt.Errorf("failed to create static vm namespace %+v", err)
+	}
+
+	// TODO: decouple cluster type with cloud provider type
+	clusterType := ""
+	switch cloudProviders {
+	case string(cloudv1alpha1.AzureCloudProvider):
+		clusterType = "aks"
+	case string(cloudv1alpha1.AWSCloudProvider):
+		clusterType = "eks"
+	}
+
+	_ = os.Setenv("KUBECONFIG", kubeconfig)
+	cmd := exec.Command("./ci/generate-agent-config.sh", "--cluster-type", clusterType, "--antrea-version", antreaVersion,
+		"--target-dir", dir, "--ns", ns.Name)
+	outputBytes, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to generate antrea agent kubeconfigs %+v: %s", err, string(outputBytes))
+	}
+
+	absPath, err := filepath.Abs("./hack/install-vm-agent-wrapper.sh")
+	if err != nil {
+		return err
+	}
+
+	_ = os.Setenv("TF_VAR_with_agent", "true")
+	_ = os.Setenv("TF_VAR_antrea_agent_k8s_config", dir+"/antrea-agent.kubeconfig")
```

Comment:
dont you want to do any validation if antrea-agent.kubeconfig or antrea-agent.antrea.kubeconfig files are present in that temp dir

Reply:
os.MkdirTemp guarantees a unique new directory under /tmp, otherwise it will error out

Reply:
I understand, I am saying, what if script doesnt generate the file and no error is returned. So safe guard, we can check for file presence

Reply:
Ah, in that case, terraform will return error when not able to find the file, causing test error out.